### PR TITLE
ISDB-294-295 C100 Part 3a-3e

### PR DESCRIPTION
**Pull request by adetolacog**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/resources/application-c100-form.yaml [hunk 1]
```diff
@@ -5,6 +5,16 @@ uk:
...
+              - existingCase_onEmergencyProtection_Care_or_supervisioNorder
+              - exemption_to_attend_MIAM
+              - familyMember_Intimation_on_No_MIAM
+              - attended_MIAM
+              - NoMIAM_domesticViolence
+              - NoMIAM_childProtectionConcerns
+              - NoMIAM_Urgency
+              - NoMIAM_PreviousAttendance
+              - NoMIAM_otherReasons
               - child1_firstName
```
Comment (hmcts-kodatala) [inline src/main/resources/application-c100-form.yaml:18]: pls move to optional fileds

File: src/test/resources/application-test-C100-form.yaml [hunk 2]
```diff
@@ -5,6 +5,16 @@ uk:
...
+              - previous_or_ongoingProceeding
+              - existingCase_onEmergencyProtection_Care_or_supervisioNorder
+              - exemption_to_attend_MIAM
+              - familyMember_Intimation_on_No_MIAM
+              - attended_MIAM
+              - NoMIAM_domesticViolence
+              - NoMIAM_childProtectionConcerns
+              - NoMIAM_Urgency
+              - NoMIAM_PreviousAttendance
+              - NoMIAM_otherReasons
```
Comment (hmcts-kodatala) [inline src/test/resources/application-test-C100-form.yaml:17]: need to move optional fileds

File: src/test/resources/application-test-c100-validation.yaml [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            mandatoryFields:
+              - previous_or_ongoingProceeding
+              - existingCase_onEmergencyProtection_Care_or_supervisioNorder
+              - exemption_to_attend_MIAM
+              - familyMember_Intimation_on_No_MIAM
+              - attended_MIAM
+              - NoMIAM_domesticViolence
+              - NoMIAM_childProtectionConcerns
+              - NoMIAM_Urgency
+              - NoMIAM_PreviousAttendance
```
Comment (hmcts-kodatala) [inline src/test/resources/application-test-c100-validation.yaml:16]: Need to remove this file all places

File: src/test/resources/request/bulk-scan-c100-validation-input.json [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+    },
+    {
+      "name": "existingCase_onEmergencyProtection_Care_or_supervisioNorder",
+      "value": "No"
+    },
+    {
+      "name": "familyMember_Intimation_on_No_MIAM",
+      "value": "No"
+    },
+    {
```
Comment (hmcts-kodatala) [inline src/test/resources/request/bulk-scan-c100-validation-input.json:19]: need to remove this files

File: config/pmd/ruleset.xml [hunk 5]
```diff
@@ -4,122 +4,130 @@
...
          xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 https://pmd.sourceforge.io/ruleset_2_0_0.xsd">
 
-  <description>HMCTS PMD rule set</description>
+    <description>HMCTS PMD rule set</description>
 
-  <rule ref="category/java/bestpractices.xml">
-    <exclude name="GuardLogStatement"/>
-    <exclude name="JUnitTestContainsTooManyAsserts"/>
-    <exclude name="JUnitAssertionsShouldIncludeMessage"/>
-    <exclude name="JUnitTestsShouldIncludeAssert"/>
```
Comment (hmcts-kodatala) [inline config/pmd/ruleset.xml:13]: Pls revert this files .

**Review by hmcts-kodatala — COMMENTED** [review]
Pls correct all review comments

File: src/main/resources/application-c100-form.yaml (hunk 1 above)
Comment (adetolacog) [reply src/main/resources/application-c100-form.yaml:18]: Removed conditional mandatory fields to Optional field group

File: src/test/resources/application-test-c100-validation.yaml (hunk 3 above)
Comment (adetolacog) [reply src/test/resources/application-test-c100-validation.yaml:16]: Done. Conditional mandatory fields now moved to Optional field group.

File: src/test/resources/application-test-C100-form.yaml (hunk 2 above)
Comment (adetolacog) [reply src/test/resources/application-test-C100-form.yaml]: Done. All conditional mandatory fields now moved to Option Field group

**Review by hmcts-kodatala — APPROVED** [review]
OK

**Review by saravanakumar-palanichamy — APPROVED** [review]

**Pull request merged**